scalafix-core/shared/src/main/scala/scalafix/internal/config/DisableConfig.scala

```diff
@@ -2,18 +2,67 @@ package scalafix
 package internal.config
 
 import metaconfig._
-import org.langmeta.Symbol
-
-import scalafix.internal.util.SymbolOps
 import metaconfig.annotation.{Description, ExampleValue}
 import metaconfig.generic.Surface
+import org.langmeta.Symbol
+
+import scala.util.matching.Regex
+
+case class DisabledSymbol(
+    @Description("Symbol to ban.")
+    symbol: Option[Symbol.Global],
+    @Description("Custom message.")
+    message: Option[String],
+    @Description("Custom id for error messages.")
+    id: Option[String],
+    @Description(
+      "When true all overrides of the symbol are also banned. Works only with symbol field, not regex. Doesn't work in unlessSynthetic mode.")
```

Comment:
Can you please break long sentences like this down into multiple string literals?

```scala
 +      "When true all overrides of the symbol are also banned. " +
          "Works only with symbol field, not regex. " +
          "Doesn't work in unlessSynthetic mode.")

Reply:
s/unlessSynthetic/ifSynthetic/

Reply:
I propose s/banHierarchy/banOverrides` and alternative phrase the description to

> If true, additionally ban references to methods that override of this symbol.
> Works only with the `symbol` option, not `regex`.

I would then guard against mixing this option with regex in ConfDecoder.